### PR TITLE
Add changeset tooling

### PR DESCRIPTION
**Pull request by stigi**
## Change description

This PR adds tooling around managing change sets, by employing https://github.com/changesets/changesets

The source of truth for the project version now relies in `package.json` and is propagated via a script to all other places (`Readme.md`, swift sources).

I also imported the last changeset from the current version 2.0.0 as a baseline and added pending changesets for work that was done in #38 and that will go out in a major release.

A few things went into this to make it work:
- Added NPM configuration and the `@changeset` dependencies
- Initialized the changeset config with the default values
- Added a wrapper around `yarn changeset` that also calls the `update-version.ts` script to update the version accross the repo.
- Cocoapods is fine with loading the version directly via JSON import from `package.json`

## Test Plan

Check out this branch, create an example changeset via `yarn changeset` or `yarn changeset add`. Bump the version via `yarn changeset version`. The version should be updated in `package.json`, `Readme.md` as well `MagicBellClient.swift`.

Also take a look at https://github.com/magicbell/magicbell-swift/pull/42 which was created by the workflow that's part of this PR.

## Type of Change

- [ ] Bug fix      
- [ ] Feature      
- [ ] Enhancement

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (stigi) [inline .changeset/config.json:1]: ⚠️ Didn't look much into them, so please advise for changes.

File: .changeset/config.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "$schema": "https://unpkg.com/@changesets/config@3.0.3/schema.json",
+  "changelog": "@changesets/cli/changelog",
+  "commit": false,
+  "fixed": [],
+  "linked": [],
+  "access": "restricted",
+  "baseBranch": "main",
+  "updateInternalDependencies": "patch",
+  "ignore": []
+}
```
Comment (stigi) [inline .changeset/config.json:11]: ⚠️ Didn't look much into them, so please advise for changes.

Comment (stigi) [reply .changeset/config.json:11]: @smeijer I was considering adding `privatePackages: { version: true, tag: true }` but I have to admit I don't fully understand its effects, even after reading the docs: https://github.com/changesets/changesets/blob/main/docs/versioning-apps.md 😊

Comment (smeijer) [reply .changeset/config.json:11]: Not entirely sure either, I think things look good like this. Let's just try it, and fix later if it's not?

**Review by smeijer — APPROVED** [review]

**Comment by stigi** [comment]
@smeijer heads up that I did a small, but significant, change in https://github.com/magicbell/magicbell-swift/pull/39/commits/153640ac52fee790a89b79d69ed2d3092a011078

The workflow now invokes `yarn changeset version` instead of calling the JS function directly (see [action code](https://github.com/changesets/action/blob/50750fa876cc1e54c7cb972db5e2f7271fc53d99/src/run.ts#L332-L343)). This makes sure that the version numbers are correctly replaced everywhere.

**Pull request merged**